### PR TITLE
Added regtest support

### PR DESCRIPTION
**Pull request by schulterklopfer**

### CODE REVIEW

**Comment by Kexkey** [comment]
`generate` is deprecated!  :-o

Ok I had to `generatetoaddress` blocks for sparkwallet/clightning to be able to run.  We need to document that, or call `generatetoaddress` during start if there's less than 101 blocks in the regtest blockchain.

Oh, fyi, I called `getnewaddress` on spending01.dat and generated coinbase to that address.

Besides that, lgtm!

Please rebase on dev.

**Comment by schulterklopfer** [comment]
will rebase. dunno what to do about the other stuff :)

**Comment by Kexkey** [comment]
Nothing, it's just good to know that c-lightning/sparkwallet won't work if there's no blocks mined in the regtest blockchain.

File: cyphernodeconf_docker/templates/bitcoin/bitcoin.conf [hunk 1]
```diff
@@ -32,6 +35,13 @@ test.wallet=xpubwatching01.dat
 test.wallet=spending01.dat
 test.wallet=ln01.dat
 test.rpcbind=0.0.0.0
+<% } else if ( net === 'regtest' ) { %>
+regtest.wallet=watching01.dat
+regtest.wallet=xpubwatching01.dat
+regtest.wallet=spending01.dat
+regtest.wallet=ln01.dat
+regtest.rpcbind=0.0.0.0
+regtest.rpcport=18443
```
Comment (Kexkey) [inline cyphernodeconf_docker/templates/bitcoin/bitcoin.conf]: Why do you specify the port number for regtest but not for the other chains?  Should we just use the default one?  Or specify them for testnet and mainnet?

Comment (schulterklopfer) [reply cyphernodeconf_docker/templates/bitcoin/bitcoin.conf]: whats the default regtest port? :D

**Pull request merged**